server/src/library/libraryUtils.ts

```diff
@@ -70,7 +92,24 @@ function getOrCreateLibraryDefinitions(
         definitionsByLocale.set(locale, mapped);
     }
 
-    return PQP.MapUtils.assertGet(definitionsByLocale, locale);
+    let result: Map<string, PQLS.Library.TLibraryDefinition> = PQP.MapUtils.assertGet(definitionsByLocale, locale);
+
+    // we got module library for this definition call
+    if (currentModuleLibraryJson.length) {
+        // make sure we do immutable operation over here
+        const base: Map<string, PQLS.Library.TLibraryDefinition> = result;
+        result = new Map();
+
+        for (const [key, value] of base) {
```

Comment:
nitpick: `base.entries()`

Reply:
we replace this by merely shallow copying at level of externalTypeResolver and libraryDefinitions